platform-controller/src/main/java/org/hobbit/controller/PlatformController.java

```diff
@@ -641,9 +719,19 @@ public void handleFrontEndCmd(byte bytes[], String replyTo, BasicProperties repl
         LOGGER.debug("Finished handling of front end request.");
     }
 
-    protected Model getChallengeFromUri(String challengeUri) {
-        // get experiments from the challenge
-        String query = SparqlQueries.getChallengeGraphQuery(challengeUri, Constants.CHALLENGE_DEFINITION_GRAPH_URI);
+    /**
+     * Retrieves model for the given challenge from the given graph (or without
+     * selecting a certain graph if the graphUri is {@code null}).
+     * 
+     * @param challengeUri
+     *            the URI for which the model should be retrieved
+     * @param graphUri
+     *            the URI from which the data should be retrieved or {@code null} if
+     *            all graphs should be taken into account.
+     * @return the RDF model of the challenge
+     */
+    protected Model getChallengeFromUri(String challengeUri, String graphUri) {
```

Comment:
Is it safe to discard one argument method in favor of two argument method?

Reply:
What exactly do you mean with safe?

The reason for this change is that the original method was only requesting information from the challenge definition graph. It was used in two different methods of the platform. 
However, it turned out that one of them has to query all graphs while the other has to query only the challenge def. graph. This solution makes sure that there is still only one method for retrieving a challenge model which can be used in both situations.